enhancements/testing/ephemeral-agent-installer-e2e-tests.md

```diff
@@ -0,0 +1,275 @@
+---
+title: ephemeral-agent-installer-e2e-tests
```

Comment:
So you're basically suggesting, that this time, this test framework, unlike all others, will be error prone, super easy to read and follow, debugable, easy to extend, collect result and operate. Of course not like others.
BTW, test infra started with the same goal. And dev scripts. And prow. And many others test frameworks as well.
The problem comes when you have to add features to it - static ips, ipxe, pxe?, vsphere, add workers, use cluster0 as a hub.... I wonder how then it will suit the suggested design pattern.
And, what will happen in a year, when someone else will come and suggest 4th test framework, because the other 3 are not readable, debugguble, too complex....